java/src/jmri/Manager.java

```diff
@@ -238,4 +238,16 @@
     @CheckReturnValue
     @Nonnull
     public String getBeanTypeHandled();
+
+    /**
+     * Enforces, and as a user convenience converts to, the standard form for a system name
+     * for the NamedBeans handled by this manager.
+     *
+     * @param inputName System name to be normalized
+     * @throws BadSystemNameException If the inputName can't be converted to normalized form
+     * @return A system name in standard normalized form 
+     */
+    @CheckReturnValue
+    public @Nonnull String normalizeSystemName(@Nonnull String inputName) throws BadSystemNameException;
+    public class BadSystemNameException extends IllegalArgumentException {}
```

Comment:
Can I suggest that this exception be in NamedBean along with the BadUserNameException?